src/ports/catalog/component.ts

```diff
@@ -20,10 +21,12 @@ import { getLatestSubgraphSchema } from './queries'
 
 export function createCatalogComponent(options: {
   database: IPgComponent
+  favoritesComponent: IFavoritesComponent
+  isFavoritesEnabled: boolean
 }): ICatalogComponent {
-  const { database } = options
+  const { database, favoritesComponent, isFavoritesEnabled } = options
 
-  async function fetch(filters: CatalogFilters) {
+  async function fetch(filters: CatalogOptions) {
```

Comment:
In the other ports that also enhance the data with the picks stats, we deconstructed the filters so we have `{ pickedBy, ...filters}` and use the "real" filters to query what we need.